rkt/fetch_keys.go

```diff
@@ -0,0 +1,237 @@
+	}
+
+	return kls, nil
+}
+
+// metaDiscoverPubKeyLocations discovers the public key through ACDiscovery by applying prefix as an ACApp
+func metaDiscoverPubKeyLocations(prefix string, allowHTTP bool, debug bool) ([]string, error) {
+	app, err := discovery.NewAppFromString(prefix)
+	if err != nil {
+		return nil, err
+	}
+
+	ep, attempts, err := discovery.DiscoverPublicKeys(*app, allowHTTP)
+	if err != nil {
+		return nil, err
+	}
+
+	if debug {
+		for _, a := range attempts {
+			stderr("meta tag 'ac-discovery-pubkeys' not found on %s: %v", a.Prefix, a.Error)
+		}
+	}
+
+	return ep.Keys, nil
+}
+
+// getPubKey retrieves a public key (if remote), and verifies it's a gpg key
+func getPubKey(location string, allowHTTP bool) (*os.File, error) {
+	u, err := url.Parse(location)
+	if err != nil {
+		return nil, err
+	}
+
+	switch u.Scheme {
+	case "":
+		return os.Open(location)
+	case "http":
+		if !allowHTTP {
+			return nil, fmt.Errorf("--insecure-allow-http required for http URLs")
+		}
+		fallthrough
+	case "https":
+		return downloadKey(u.String())
+	}
+
+	return nil, fmt.Errorf("only http and https urls supported")
+}
+
+// downloadKey retrieves the file, storing it in a deleted tempfile
+func downloadKey(url string) (*os.File, error) {
+	tf, err := ioutil.TempFile("", "")
+	if err != nil {
+		return nil, fmt.Errorf("error creating tempfile: %v", err)
+	}
+	os.Remove(tf.Name()) // no need to keep the tempfile around
+
+	defer func() {
+		if err != nil {
+			tf.Close()
+		}
+	}()
+
+	res, err := http.Get(url)
+	if err != nil {
+		return nil, fmt.Errorf("error getting key: %v", err)
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("bad HTTP status code: %d", res.StatusCode)
+	}
+
+	if _, err := io.Copy(tf, res.Body); err != nil {
+		return nil, fmt.Errorf("error copying key: %v", err)
+	}
+
+	tf.Seek(0, os.SEEK_SET)
+
+	return tf, nil
+}
+
+// addKeys adds the keys listed in pkls at prefix
+func addKeys(pkls []string, prefix string, allowHTTP bool, forceAccept bool, allowOverride bool) error {
+	ks := getKeystore()
+
+	for _, pkl := range pkls {
+		pk, err := getPubKey(pkl, allowHTTP)
+		if err != nil {
+			return fmt.Errorf("error accessing key: %v", err)
+		}
+		defer pk.Close()
+
+		exists, err := ks.TrustedKeyPrefixExists(prefix, pk)
+		if err != nil {
+			return fmt.Errorf("error reading key: %v", err)
+		}
+		err = displayKey(prefix, pkl, pk)
+		if err != nil {
+			return fmt.Errorf("error displayKey key: %v", err)
+		}
+		if exists && !allowOverride {
+			stderr("Key %q already in the keystore", pkl)
+			return nil
+		}
+
+		if !forceAccept {
+			accepted, err := reviewKey()
+			if err != nil {
+				return fmt.Errorf("error reviewing key: %v", err)
+			}
+			if !accepted {
+				stderr("Not trusting %q", pkl)
+				continue
+			}
+		}
+
+		if forceAccept {
+			stderr("Trusting %q for prefix %q without fingerprint review.", pkl, prefix)
```

Comment:
Ok, I did a bit of git-grepping here. So looks like `stdout` is used only for expected output of a command (like printing a pod's status in `rkt status`).

Nevermind then.